### PR TITLE
Fix Lock and Unlock system

### PR DESCRIPTION
**Pull request by panleone**
## Abstract

Fix and simplify the lock/unlock system, main changes are:

- Make lock/unlock work again
- Move `decryptWallet` to the Wallet class
- Move `wipePrivateData` to Dashboard.vue and rename it to `lockWallet` 
- Remove unused variables from index.js and global.js
- Add JsDoc to make things more clear

### CODE REVIEW

**Review by Liquid369 — APPROVED** [review]
tACK dc3f5c348f5af1f6fe538297aa7cc2083cb702cd
Looks to be working as intended

**Review by JSKitty — APPROVED** [review]
**tACK** dc3f5c348f5af1f6fe538297aa7cc2083cb702cd, glad this was caught, no idea how I missed it. Tested `master` and then compared with dc3f5c348f5af1f6fe538297aa7cc2083cb702cd, working well and as expected on production.

File: scripts/wallet.js [hunk 1]
```diff
@@ -250,6 +251,31 @@ export class Wallet {
         return !!this.#masterKey;
     }
 
+    /**
+     * Check if the current encrypted keyToBackup can be decrypted with the given password
+     * @param {string} strPassword
+     * @return {Promise<boolean>}
+     */
+    async decryptWallet(strPassword = '') {
```
Comment (Duddino) [inline scripts/wallet.js:259]:
```suggestion
    async decrypt(strPassword = '') {
```

File: scripts/wallet.js [hunk 2]
```diff
@@ -250,6 +251,31 @@ export class Wallet {
...
+        // Check if there's any encrypted WIF available
+        const database = await Database.getInstance();
+        const { encWif: strEncWIF } = await database.getAccount();
+        if (!strEncWIF || strEncWIF.length < 1) return false;
+
+        // Prompt to decrypt it via password
+        const strDecWIF = await decrypt(strEncWIF, strPassword);
+        if (!strDecWIF || strDecWIF === 'decryption failed!') {
+            if (strDecWIF)
+                return createAlert('warning', ALERTS.INCORRECT_PASSWORD, 6000);
```
Comment (Duddino) [inline scripts/wallet.js:269]:
```suggestion
                return false;
```
`Wallet.decrypt` should work like `Wallet.encrypt` (in #248), it should return a boolean without alerting

File: scripts/dashboard/Dashboard.vue [hunk 3]
```diff
@@ -179,43 +180,27 @@ async function importWallet({ type, secret, password = '' }) {
...
-
 /**
  * Encrypt wallet
  * @param {string} password - Password to encrypt wallet with
  * @param {string} [currentPassword] - Current password with which the wallet is encrypted with, if any
  */
 async function encryptWallet(password, currentPassword = '') {
     if (await hasEncryptedWallet()) {
-        if (!(await decryptWallet(currentPassword))) return;
+        if (!(await wallet.decryptWallet(currentPassword))) return;
```
Comment (Duddino) [inline scripts/dashboard/Dashboard.vue:203]:
```suggestion
        if (!(await wallet.decryptWallet(currentPassword))) return createAlert(...);
```

**Comment by panleone** [comment]
conflicts solved and review tackled

File: scripts/wallet.js [hunk 4]
```diff
@@ -250,7 +251,27 @@ export class Wallet {
...
+     * @return {Promise<boolean>}
+     */
+    async decrypt(strPassword) {
+        // Check if there's any encrypted WIF available
+        const database = await Database.getInstance();
+        const { encWif: strEncWIF } = await database.getAccount();
+        if (!strEncWIF || strEncWIF.length < 1) return false;
+
+        const strDecWIF = await decrypt(strEncWIF, strPassword);
+        return !strDecWIF ? false : true;
```
Comment (Duddino) [inline scripts/wallet.js:266]:
```suggestion
        return !!strDecWIF;
```

File: scripts/wallet.js [hunk 5]
```diff
@@ -250,7 +251,27 @@ export class Wallet {
         return !!this.#masterKey;
     }
 
-    async encryptWallet(strPassword) {
+    /**
+     * Check if the current encrypted keyToBackup can be decrypted with the given password
+     * @param {string} strPassword
+     * @return {Promise<boolean>}
+     */
+    async decrypt(strPassword) {
```
Comment (Duddino) [inline scripts/wallet.js:259]: The name is very misleading, this doesn't `decrypt` the wallet, it just checks if the password is correct. Considering that `encrypt` does encrypt the wallet, it's a confusing interface

**Review by Duddino — CHANGES_REQUESTED** [review]

File: scripts/wallet.js (hunk 5 above)
Comment (panleone) [reply scripts/wallet.js]: changed in `checkDecryptPassword`

**Review by Duddino — APPROVED** [review]
tACK

**Review by JSKitty — APPROVED** [review]
**tACK** 95cf9cb7f85cbc00845fde92670c20f099abc5ec

**Pull request merged**